File: ser531/src/main/java/org/example/RouteData.java
package org.example;

import java.util.List;

class RouteData {
    private List<FuelStation> fuelStations;
    private List<Restaurant> restaurants;

    public RouteData(List<FuelStation> fuelStations, List<Restaurant> restaurants) {
        this.fuelStations = fuelStations;
        this.restaurants = restaurants;
    }

    public List<FuelStation> getFuelStations() {
        return fuelStations;
    }

    public void setFuelStations(List<FuelStation> fuelStations) {
        this.fuelStations = fuelStations;
    }

    public List<Restaurant> getRestaurants() {
        return restaurants;
    }

    public void setRestaurants(List<Restaurant> restaurants) {
        this.restaurants = restaurants;
    }
}

File: nearby-gas/components/RouteData.js
import React from 'react';
import FuelStation from './FuelStation'; 
import Restaurant from './Restaurant';

const RouteData = ({ fuelStations, restaurants }) => (
  <div>
    <h2>Fuel Stations</h2>
    {fuelStations.map((station, index) => (
      <FuelStation key={index} {...station} />
    ))}

    <h2>Restaurants</h2>
    {restaurants.map((restaurant, index) => (
      <Restaurant key={index} {...restaurant} />
    ))}
  </div>
);

export default RouteData;
